### ISSUE TITLE
False positive unpacking-non-sequence on classmethod + property

### ISSUE THREAD

**Issue by rominf** [PyCQA/pylint#4803]
### Bug description

```python
class E:
    @classmethod
    @property
    def p(cls):
        return 0, 42


x, y = E.p
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:8:0: E0633: Attempting to unpack a non-sequence defined at line 2 (unpacking-non-sequence)
```

### Expected behavior

No output, it's allowed in Python 3.9: https://docs.python.org/3/library/functions.html#classmethod

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.9.6 (default, Jul 16 2021, 00:00:00) 
[GCC 10.3.1 20210422 (Red Hat 10.3.1-1)
```

### OS / Environment

Fedora 33

**Comment by luinnar** [comment]
This also applies to "standard" class properties:

```python
class Test:
    OPTIONAL_TUPLE: tuple[str, str] | None = None

    def test(self) -> None:
        if self.OPTIONAL_TUPLE is None or len(self.OPTIONAL_TUPLE) != 2:
            raise ValueError("No, no, no")

        _, var1 = self.OPTIONAL_TUPLE
```

```
types.py:11:8: E0633: Attempting to unpack a non-sequence defined at line 5 (unpacking-non-sequence)
```
